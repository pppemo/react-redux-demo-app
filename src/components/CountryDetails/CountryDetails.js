import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row, Table } from 'react-bootstrap'
import styles from './CountryDetails.module.scss'

const CountryDetails = ({ countryDetails }) => {
  const { name, capital, flag, timezones, region, currencies, languages, alpha3Code } = countryDetails

  return (
    <div>
      <h1>{name}</h1>
      <Row>
        <Col sm={6}>
          <Table responsive>
            <tbody>
            <tr>
              <td><b>Alpha code</b></td>
              <td>{alpha3Code}</td>
            </tr>
            <tr>
              <td><b>Region</b></td>
              <td>{region}</td>
            </tr>
            <tr>
              <td><b>Capital</b></td>
              <td>{capital}</td>
            </tr>
            <tr>
              <td><b>Timezones</b></td>
              <td>{timezones.join(', ')}</td>
            </tr>
            <tr>
              <td><b>Currencies</b></td>
              <td>{currencies.map(currency => (
                <div key={currency.code}>{currency.name} ({currency.code})</div>
              ))}</td>
            </tr>
            <tr>
              <td><b>Languages</b></td>
              <td>{languages.map(language => language.name).join(', ')}</td>
            </tr>
            </tbody>
          </Table>
        </Col>
        <Col sm={6}>
          <div className={styles.countryFlag}>
            <img alt="Flag" src={flag}/>
          </div>
        </Col>
      </Row>
    </div>
  )
}

CountryDetails.propTypes = {
  countryDetails: PropTypes.shape({
    name: PropTypes.string,
    capital: PropTypes.string,
    flag: PropTypes.string,
    region: PropTypes.string,
    timezones: PropTypes.arrayOf(PropTypes.string),
    currencies: PropTypes.arrayOf(
      PropTypes.shape({
        code: PropTypes.string,
        name: PropTypes.string,
        symbol: PropTypes.string
      })
    ),
    languages: PropTypes.arrayOf(
      PropTypes.shape({
        iso639_1: PropTypes.string,
        iso639_2: PropTypes.string,
        name: PropTypes.string,
        nativeName: PropTypes.string
      })
    ),
    alpha3Code: PropTypes.string
  }).isRequired
}

export default CountryDetails
