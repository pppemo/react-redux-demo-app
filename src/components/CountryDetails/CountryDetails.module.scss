.countryFlag {
  text-align: center;

  img {
    width: 60%;
    border: 1px solid black;
  }
}
