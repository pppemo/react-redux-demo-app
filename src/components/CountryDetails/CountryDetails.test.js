import TestRenderer from 'react-test-renderer'
import React from 'react'
import CountryDetails from './CountryDetails'

describe('CountryDetails component', () => {
  it('should render details', () => {
    const details = {
      currencies: [
        {
          code: "FKP",
          name: "Falkland Islands pound",
          symbol: "Â£"
        }
      ],
      languages: [
        {
          iso639_1: "en",
          iso639_2: "eng",
          name: "English",
          nativeName: "English"
        }
      ],
      flag: "https://restcountries.eu/data/flk.svg",
      name: "Falkland Islands (Malvinas)",
      alpha3Code: "FLK",
      capital: "Stanley",
      region: "Americas",
      timezones: [
        "UTC-04:00"
      ]
    }

    const tree = TestRenderer
      .create(<CountryDetails countryDetails={details} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
