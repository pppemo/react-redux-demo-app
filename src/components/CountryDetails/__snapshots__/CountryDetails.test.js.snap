// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetails component should render details 1`] = `
<div>
  <h1>
    Falkland Islands (Malvinas)
  </h1>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table"
        >
          <tbody>
            <tr>
              <td>
                <b>
                  Alpha code
                </b>
              </td>
              <td>
                FLK
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Region
                </b>
              </td>
              <td>
                Americas
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Capital
                </b>
              </td>
              <td>
                Stanley
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Timezones
                </b>
              </td>
              <td>
                UTC-04:00
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Currencies
                </b>
              </td>
              <td>
                <div>
                  Falkland Islands pound
                   (
                  FKP
                  )
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Languages
                </b>
              </td>
              <td>
                English
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="countryFlag"
      >
        <img
          alt="Flag"
          src="https://restcountries.eu/data/flk.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
